plugins/scaffolder-backend/src/lib/templater/index.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CookieCutter } from './cookiecutter';
+
+export interface RequiredTemplateValues {
+  componentId: string;
+}
+export interface TemplaterRunOptions {
+  directory: string;
+  values: RequiredTemplateValues & object;
+}
+
+export abstract class TemplaterBase {
```

Comment:
Any particular reason for making these abstract classes btw? Usually just end up with plain types for these. Does it make type inference of methods work?

Reply:
I guess I thought it just makes it easier for people to implement their own when they must adhere to a particular contract. To me it makes more sense that you need to implement something that takes `x` and returns `y` as an implentation rather than extending another class as we cant really provide any useful functions for people to reusing when were talking repository or templating libraries. If that makes sense?

Reply:
But can just as easily do `implement type`